src/icrc2-minter/src/tasks.rs

```diff
@@ -121,14 +121,32 @@ impl BridgeTask {
             ));
         };
 
-        let logs = BridgeEvent::collect_logs(
+        let logs = match BridgeEvent::collect_logs(
```

Comment:
here it is required to paginate to queries if more than X blocks have to be fetched.
E.g., pseudo code:
```
let to_block = from_block + X;

///  loop until there are more than X blocks in the response
while query_logs.query().blocks >= X {
    ...
}
```